plugin-server/src/main/ingestion-queues/session-recording/utils.ts

```diff
@@ -155,17 +188,42 @@ export const parseKafkaMessage = async (
 
     const headerResult = await readTokenFromHeaders(message.headers, getTeamFn)
     const token: string | undefined = headerResult.token
-    let teamIdWithConfig: null | TeamIDWithConfig = headerResult.teamIdWithConfig
+
+    if (!token) {
+        return dropMessage('no_token_in_header')
+    }
+
+    const teamIdWithConfig: null | TeamIDWithConfig = headerResult.teamIdWithConfig
 
     // NB `==` so we're comparing undefined and null
     // if token was in the headers but, we could not load team config
     // then, we can return early
-    if (!!token && (teamIdWithConfig == null || teamIdWithConfig.teamId == null)) {
+    if (teamIdWithConfig == null || teamIdWithConfig.teamId == null) {
         return dropMessage('header_token_present_team_missing_or_disabled', {
             token: token,
         })
     }
 
+    // this has to be ahead of the payload parsing in case we start dropping traffic from older versions
+    if (!!ingestionWarningProducer && !!teamIdWithConfig.teamId) {
+        const libVersion = readLibVersionFromHeaders(message.headers)
+        const minorVersion = minorVersionFrom(libVersion)
+        if (minorVersion && minorVersion <= 74) {
```

Comment:
Perhaps worth pulling this version number to a variable explaining it was the version we introduced rate limiting in

Reply:
ended up making it a comment since there's so much context

good shout though that context was in my brain and slack messages and incident write-ups and its better here